src/Tribe/Settings_Tab.php

```diff
@@ -523,88 +541,90 @@ protected function validate_sidebar( $sidebar ) {
+	 * Adds a callback for the tab to register hooks.
 	 *
-	 * @deprecated TBD use add_all_tabs instead.
+	 * @since TBD
 	 *
-	 * @param array $all_tabs All the tabs from Tribe__Settings.
+	 * @param callable $callback The callback to add.
 	 *
-	 * @return array $all_tabs All the tabs.
+	 * @return void
 	 */
-	public function addAllTabs( $all_tabs ) {
-		_deprecated_function( __METHOD__, 'TBD', 'add_all_tabs' );
-
-		return $this->add_all_tabs( $all_tabs );
+	public function add_hook_callback( callable $callback ) {
+		$this->hook_callback = $callback;
 	}
 
-	/* Deprecated Methods */
-
 	/**
-	 * filters the fields array from Tribe__Settings
-	 * and adds the current tab's fields to it
+	 * Whether this tab has a hook callback.
 	 *
-	 * @deprecated TBD use add_fields instead.
-	 *
-	 * @param array $field the $fields from Tribe__Settings.
+	 * @since TBD
 	 *
-	 * @return array $fields the filtered fields
+	 * @return bool
 	 */
-	public function addFields( $fields ) {
-		_deprecated_function( __METHOD__, 'TBD', 'add_fields' );
-
-		return $this->add_fields( $fields );
+	public function has_hook_callback(): bool {
+		return null !== $this->hook_callback;
 	}
 
 	/**
-	 * sets whether the current tab should show the save
-	 * button or not
-	 *
-	 * @deprecated TBD use show_save_tab instead.
-	 *
-	 * @param array $no_save_tabs the $no_save_tabs from Tribe__Settings
+	 * Run the callback if there is one.
 	 *
-	 * @return array $no_save_tabs the filtered non saving tabs
-	 */
-	public function showSaveTab( $no_save_tabs ) {
-		_deprecated_function( __METHOD__, 'TBD', 'show_save_tab' );
-
-		return $this->show_save_tab( $no_save_tabs );
-	}
-
-
-	/**
-	 * Displays the content for the tab.
+	 * This will only run the callback once.
 	 *
-	 * @deprecated TBD use do_content instead.
+	 * @since TBD
 	 *
 	 * @return void
 	 */
-	public function doContent() {
-		_deprecated_function( __METHOD__, 'TBD', 'do_content' );
+	public function run_hook_callback() {
+		if ( $this->did_hook_callback ) {
+			return;
+		}
+
+		if ( ! $this->has_hook_callback() ) {
+			return;
+		}
 
-		$this->do_content();
+		$this->did_hook_callback = true;
+		call_user_func( $this->hook_callback );
 	}
 
 	/**
-	 * filters the tabs array from Tribe__Settings
-	 * and adds the current tab to it
-	 * does not add a tab if it's empty
+	 * Handle calls to methods that don't exist.
+	 *
+	 * This is how we handle deprecated methods.
 	 *
-	 * @deprecated TBD use add_tab instead.
+	 * @since TBD
 	 *
-	 * @param array $tabs the $tabs from Tribe__Settings
+	 * @param string $name      The method name.
+	 * @param array  $arguments Arguments passed to the method.
 	 *
-	 * @return array $tabs the filtered tabs
+	 * @return mixed The result of the method call.
+	 * @throws BadMethodCallException If the method does not exist.
 	 */
-	public function addTab( $tabs ) {
-		_deprecated_function( __METHOD__, 'TBD', 'add_tab' );
+	#[ReturnTypeWillChange]
```

Comment:
Is this necessary with return type `mixed`?

Reply:
Yes, because `__call()` has a default return type of `mixed`, but actually typing with `mixed` isn't supported until PHP 8.0+. I believe if we run this in PHP 8+ without a return type, it will trigger a warning.